core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -316,6 +309,50 @@ public void testSerialization() throws IOException {
         }
     }
 
+    private QB copyQuery(QB query) throws IOException {
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            query.writeTo(output);
+            try (StreamInput in = new NamedWriteableAwareStreamInput(StreamInput.wrap(output.bytes()), namedWriteableRegistry)) {
+                QueryBuilder<?> prototype = queryParserService.queryParser(query.getWriteableName()).getBuilderPrototype();
+                @SuppressWarnings("unchecked")
+                QB secondQuery = (QB)prototype.readFrom(in);
+                return secondQuery;
+            }
+        }
+    }
+
+    @Test
+    public void testEqualsAndHashcode() throws IOException {
+        QB firstQuery = createTestQueryBuilder();
+        assertFalse("query is equal to null", firstQuery.equals(null));
+        assertFalse("query is equal to incompatible type", firstQuery.equals(randomAsciiOfLengthBetween(1, 10)));
```

Comment:
right I just need a String or whatever other type. I used random because I hate using "hello" or "test", that's it. I see how it can be confusing though. What should I do?